Web/src/components/games/PandaSequence/GameDisplay.tsx

```diff
@@ -57,19 +68,45 @@ const Score = styled.h3`
   padding-top: 6px;
 `;
 
+/**
+ * This component handles both display and input mode.
+ * Display Mode: Element in sequence is animated according to "displaying" prop
+ * Input Mode: Element tapped is animated based on "selected" state
+ */
 const GameDisplay: React.FC<IProps> = ({
-  mode, secondsLeft, score, active, handleInputEvent,
+  mode, secondsLeft, score, processInput, timestep, displaying,
 }) => {
-  const onTap = (index:number) => {
-    // TODO: Animate tapped component to confirm user input
-    handleInputEvent(index);
+  const [selected, setSelected] = useState(Array(5).fill(false));
+
+  const handleTouch = (index:number) => {
+    setSelected((oldSelected) => Object.assign([], oldSelected, { [index]: true }));
+  };
+
+  const handleTouchEnd = (index:number) => {
+    processInput(index);
+    setSelected(Object.assign([], selected, { [index]: false }));
   };
 
+
   let balloons;
   if (mode === PandaSequenceMode.DISPLAY) {
-    balloons = Array.from(Array(5).keys()).map((i) => <Balloon key={i} style={{ width: active === i ? '100px' : '70px' }} />);
+    balloons = Array.from(Array(5).keys()).map((index) => (
+      <DisplayBalloon
+        key={index}
+        duration={displaying === index ? timestep : 0}
+      />
+    ));
   } else {
-    balloons = Array.from(Array(5).keys()).map((i) => <Balloon key={i} onClick={() => onTap(i)} />);
+    balloons = Array.from(Array(5).keys()).map((index) => (
+      <InputBalloon
+        onTouchStart={() => handleTouch(index)}
+        onMouseDown={() => handleTouch(index)}
+        onTouchEnd={() => handleTouchEnd(index)}
+        onMouseUp={() => handleTouchEnd(index)}
+        isselected={selected[index] ? 1 : 0}
```

Comment:
Lol why not pass in the array content itself (boolean)

Reply:
Another workaround, this time to deal with [this](https://github.com/styled-components/styled-components/issues/1198) error :/

Reply:
Ahh I see, maybe add a comment there?

Reply:
Resolved with [this fix](https://www.styled-components.com/docs/faqs#why-am-i-getting-html-attribute-warnings), as suggested!